src/platform/datapath_raw_socket.c

```diff
@@ -616,26 +745,29 @@ CxPlatDpRawParseIPv4(
         return;
     }
 
-    if (IP->Protocol == IPPROTO_UDP) {
-        uint16_t IPTotalLength;
-        IPTotalLength = CxPlatByteSwapUint16(IP->TotalLength);
+    uint16_t IPTotalLength;
+    IPTotalLength = CxPlatByteSwapUint16(IP->TotalLength);
```

Comment:
```suggestion
    uint16_t IPTotalLength = CxPlatByteSwapUint16(IP->TotalLength);
```
And I _think_ it can be `const` too.